src/other/precompile.jl

```diff
@@ -1,3136 +1,1583 @@
 # precompile.jl contains precompilation directives for all methods whose compilation
 # is triggered by running DataFrames tests and takes more than a given threshold
-# For all=true, the threshold is 0.01, for all=false it is 0.2.
+# For all=true, the threshold is 0.01, for all=false it is 0.1.
 # `precompile` calls are generated using SnoopCompile via:
-# using SnoopCompileCore
-# inf_timing = @snoopi tmin=0.01 include("test/runtests.jl")
-# using SnoopCompile
-# pc = SnoopCompile.parcel(inf_timing)
-# SnoopCompile.write("src/other/precompile_tmp.jl", pc[:DataFrames], always=true)
-# sed -i '/categorical/di' src/other/precompile_tmp.jl # Remove CategoricalArrays uses
-# sed -i '/var"/d' src/other/precompile_tmp.jl # var"" is not supported on old Julia versions
-# sed -i '/materialize!/d' src/other/precompile_tmp.jl # Work around an inference bug
-# sed -i '/setindex_widen_up_to/d' src/other/precompile_tmp.jl # Not present on Julia 1.0
-# sed -i '/restart_copyto_nonleaf!/d' src/other/precompile_tmp.jl # Not present on Julia 1.0
+#     using SnoopCompileCore
+#     inf_timing = @snoopi tmin=0.01 include("test/runtests.jl")
+#     using SnoopCompile
+#     pc = SnoopCompile.parcel(inf_timing)
+#     SnoopCompile.write("src/other/precompile_tmp.jl", pc[:DataFrames], always=true)
+# and then hand edited. The editing steps are:
+# * removing signatures with anonymous functions
+# * removing signatures with specific NamedTuples passed as arguments except `x1` as name
+#   (which is generated internally)
+# * changing Int64 to Int (to handle 32-bit architectures correctly)
+# * disabling precompilation on Julia older than 1.6
 function precompile(all=false)
+    VERSION >= v"1.6" || return nothing
```

Comment:
Maybe let's check that this works on 1.5? Better avoid regressions for people who haven't moved to 1.6 yet.
```suggestion
# * disabling precompilation on Julia older than 1.5
function precompile(all=false)
    VERSION >= v"1.5" || return nothing
```

Also, maybe the crash on 1.0.5 is due to a particular `precompile` call? IIRC the `materialize!` line above was there to remove a line that created issues on 1.0 too.

Reply:
yes, but the difference was that `materialize!` was crashing the call to `precompile` function. Here we have a problem that the crash happens after precompilation - during actual function call Julia compiler gets confused. I am OK to turn on 1.5 though. I will test if all works on 1.5.